main.tf

```diff
@@ -446,6 +446,7 @@ resource "azurerm_kubernetes_cluster" "main" {
 
     content {
       log_analytics_workspace_id = local.log_analytics_workspace.id
+      msi_auth_for_monitoring_enabled = var.msi_auth_for_monitoring_enabled
```

Comment:
This is already present in provider version 3.47.0:
https://github.com/hashicorp/terraform-provider-azurerm/pull/20757

No need to bump versions. All good.